lib/ansible/modules/package_facts.py

```diff
@@ -408,7 +410,8 @@ def main():
 
     # start work
     global module
-    module = AnsibleModule(argument_spec=dict(manager={'type': 'list', 'default': ['auto']},
+    module = AnsibleModule(argument_spec=dict(manager={'type': 'list', 'elements': 'str', 'default': ['auto'],
+                                              'choices': ['apt', 'auto', 'pacman', 'pkg', 'portage', 'rpm', 'apk']},
```

Comment:
This is the only change I'm not really sure whether this behavior is wanted. Right now, even though the documentation forces you to stick to this list of values, you can specify something else. The code below valides this option, and potentially allows new values that are not yet listed in `choices`, as long as support for them has been implemented.

Also, right now, `PACMAN` for example is also accepted, and converted to `pacman` by the module (https://github.com/ansible/ansible/blob/c6ce08bb5d93f7d625905d031a4567d54f00a961/lib/ansible/modules/package_facts.py#L419). With this change, this is no longer possible.

Reply:
It seems that ignoring `doc-choices-do-not-match-spec` is reasonable because of the module implementation, I'm going to ignore it in this PR.

Reply:
You still need to remove the `choices` from `argument_spec`.